lib/core/src/getRodsEnv.cpp

```diff
@@ -139,12 +142,19 @@ extern "C" {
         createRodsEnvDefaults( &rodsEnvArg );
     }
 
-    static
-    int capture_string_property(
-        const std::string&             _key,
-        char*                          _val ) {
+    static int capture_string_property(const std::string& _key, char* _val, std::size_t _val_size)
```

Comment:
Since the function is used in a unit test and declared in a header file, it should probably not be defined as static, as that typically makes the function unavailable outside its translation unit.